bin/database.py

```diff
@@ -1054,6 +1057,67 @@ def upgrade_to_29():
     users = config.db.users.find({})
     process_cursor(users, upgrade_to_29_closure)
 
+def upgrade_to_30_closure_analysis(coll_item, coll):
+    analyses = coll_item.get('analyses', None)
+
+    if analyses is not None:
+        for analysis_ in analyses:
+            files = analysis_.get('files', [])
+            for file_ in files:
+                if 'created' not in file_:
+                    file_['created'] = analysis_.get('created', datetime.datetime(1970, 1, 1))
+        result = coll.update_one({'_id': coll_item['_id']}, {'$set': {'analyses': analyses}})
+        if result.modified_count == 1:
+            return True
+        else:
+            return "File timestamp creation failed for:" + str(session) + '/analyses' + str(analysis_) + '/files' + str(file_)
+
+def upgrade_to_30_closure_coll(coll_item, coll):
+    files = coll_item.get('files', [])
+    for file_ in files:
+        if 'created' not in file_:
+            file_['created'] = coll_item.get('created', datetime.datetime(1970, 1, 1))
+    result = coll.update_one({'_id': coll_item['_id']}, {'$set': {'files': files}})
+    if result.modified_count == 1:
+        return True
+    else:
+        return "File timestamp creation failed for:" + str(session) + '/files' + str(file_)
+
+
+def upgrade_to_30():
+    """
+    scitran/core issue #759
+
+    give created timestamps that are missing are given based on the parent object's timestamp
+    """
+    config.db.projects.insert_one({'created': datetime.datetime.utcnow(), 'files': [{'name': "file1"}]})
+    config.db.sessions.insert_one({'created': datetime.datetime.utcnow(), 'files': [{'name': "file1"}]})
+    config.db.acquisitions.insert_one({'created': datetime.datetime.utcnow(), 'files': [{'name': "file1"}]})
+
+    config.db.projects.insert_one({'files': [{'name': "file2"}]})
+    config.db.collections.insert_one({'analyses': [{'files': [{'name': 'file1'}]}]})
+    config.db.collections.insert_one({'analyses': [{'created': datetime.datetime.utcnow(), 'files': [{'name': 'file2'}]}]})
+
+    cursor = config.db.collections.find({'analyses': {'$exists': True},
+                                                       'analyses.files.created': {'$exists': False}})
+    process_cursor(cursor, upgrade_to_30_closure_analysis, config.db.collections)
```

Comment:
Just an existing pattern in the API is to pass the name of the collection, not a reference to the collection itself when doing work like this. For example:

`process_cursor(cursor, upgrade_to_30_closure_analysis, content='collections')`

and then in the closure:
`config.db[context].insert(...)`

But that is just a style choice and doesn't really affect the result of the code.